src/liballoc/collections/btree/search.rs

```diff
@@ -54,12 +54,16 @@ where
     Q: Ord,
     K: Borrow<Q>,
 {
-    for (i, k) in node.keys().iter().enumerate() {
-        match key.cmp(k.borrow()) {
-            Ordering::Greater => {}
-            Ordering::Equal => return (i, true),
-            Ordering::Less => return (i, false),
+    let len = node.len();
+    if len > 0 {
+        // make sure that `!node.is_shared_root()`
```

Comment:
To be clear, this comment refers to the `len > 0`, right? I read it as referring to the `for` *after* the comment and was confused for a second. Maybe put it in the same line as the `if`?

Reply:
That's where I wrote it, but formatting won't stand for it. I could do it one line above, or (eeeuw) in a /* */

Reply:
Ah, starting to fight `rustfmt` already...

Yeah but it above then, please -- and add an explicit reference ("checking length to make sure that").